packages/block-editor/src/hooks/border-color.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { getBlockSupport, hasBlockSupport } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ColorGradientControl from '../components/colors-gradients/control';
+import {
+	getColorClassName,
+	getColorObjectByColorValue,
+} from '../components/colors';
+import useEditorFeature from '../components/use-editor-feature';
+import { BORDER_SUPPORT_KEY } from './border';
+import { cleanEmptyObject } from './utils';
+
+const BORDER_COLOR_SUPPORT_KEY = 'color';
+const EMPTY_ARRAY = [];
+
+/**
+ * Inspector control panel containing the border color related configuration.
+ *
+ * There is deliberate overlap between the colors and borders block supports
+ * relating to border color. It can be argued the border color controls could
+ * be included within either, or both, the colors and borders panels in the
+ * inspector controls. If they share the same block attributes it should not
+ * matter.
+ *
+ * @param  {Object} props Block properties.
+ * @return {WPElement}    Border color edit element.
+ */
+export function BorderColorEdit( props ) {
+	const {
+		attributes: { borderColor, style },
+		setAttributes,
+	} = props;
+	const colors = useEditorFeature( 'color.palette' ) || EMPTY_ARRAY;
+
+	const disableCustomColors = ! useEditorFeature( 'color.custom' );
+	const disableCustomGradients = ! useEditorFeature( 'color.customGradient' );
+
+	if ( useIsBorderColorDisabled( props ) ) {
+		return null;
+	}
+
+	const onChangeColor = ( value ) => {
+		const colorObject = getColorObjectByColorValue( colors, value );
+		const newStyle = {
+			...style,
+			border: {
+				...style?.border,
+				color: colorObject?.slug ? undefined : value,
+			},
+		};
+
+		const newNamedColor = colorObject?.slug ? colorObject.slug : undefined;
```

Comment:
`colorObject?.slug` will return `undefined' if there is no 'slug' property. I believe this ternary is not needed here.

Reply:
I was under the impression the colors were coming directly from the theme.json. Given it could be supplied by a theme developer and the slug accidentally left as an empty string it seemed prudent to ensure undefined. It's also consistent with `color.js` and `with-colors.js` at present.

Happy to change it of course if it should be simplified to `colorObject?.slug`

Reply:
ah, if this is a case, I think it would be good to add a comment as this piece of code could be misleading

Reply:
I've added a comment regarding enforcing undefined in case a theme gives us an empty slug value in the color preset. Hopefully that might assist in making things clearer.

The suggested code snippet would actually introduce a bug I'm afraid. We need to set the attribute to undefined to clear any previous named color selection.